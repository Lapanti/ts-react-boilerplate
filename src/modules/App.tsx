import * as React from 'react';
import { Router, Route, browserHistory } from 'react-router';

/* tslint:disable:no-any */
const App: React.StatelessComponent<any> = props => (
    /* tslint:enable:no-any */
    <div className="app-base">
        {props.children}
    </div>
);

export default App;
