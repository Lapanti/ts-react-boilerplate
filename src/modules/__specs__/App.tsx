import * as React from 'react';
import { shallow } from 'enzyme';
import createHistory from 'history/createBrowserHistory';
import { App } from '../App';

describe('App', () => {
    const index = (
        <App
            match={{ params: undefined, isExact: true, path: '', url: '' }}
            location={{ pathname: '', search: '', state: {}, hash: '', key: '' }}
            history={createHistory()}
        />
    );
    const notFound = (
        <App
            match={{ params: undefined, isExact: false, path: '', url: '' }}
            location={{ pathname: '', search: '', state: {}, hash: '', key: '' }}
            history={createHistory()}
        />
    );

    it('should render correctly', () => {
        expect(shallow(index)).toMatchSnapshot();
        expect(shallow(notFound)).toMatchSnapshot();
    });
});
