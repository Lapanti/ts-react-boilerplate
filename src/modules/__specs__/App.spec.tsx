import * as React from 'react';
import { shallow } from 'enzyme';
import { StaticRouter } from 'react-router-dom';
import createHistory from 'history/createBrowserHistory';

import { App } from '../App';

describe('App', () => {
  const index = (
    <StaticRouter context={{}}>
      <App />
    </StaticRouter>
  );
  const notFound = (
    <StaticRouter context={{}}>
      <App />
    </StaticRouter>
  );

  it('should render correctly', () => {
    expect(shallow(index)).toMatchSnapshot();
    expect(shallow(notFound)).toMatchSnapshot();
  });
});
