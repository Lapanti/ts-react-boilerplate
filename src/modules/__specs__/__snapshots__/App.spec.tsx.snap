// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Component />
</Router>
`;

exports[`App should render correctly 2`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Component />
</Router>
`;
